process/rating/ratingsData.go

```diff
@@ -96,15 +114,130 @@ func NewRatingsData(args RatingsDataArg) (*RatingsData, error) {
+			roundTimeMillis:                 rd.roundDurationInMilliseconds,
+			startRating:                     rd.ratingsSetup.General.StartRating,
+			maxRating:                       rd.ratingsSetup.General.MaxRating,
+			hoursToMaxRatingFromStartRating: rd.ratingsSetup.MetaChain.HoursToMaxRatingFromStartRating,
+			proposerDecreaseFactor:          rd.ratingsSetup.MetaChain.ProposerDecreaseFactor,
+			validatorDecreaseFactor:         rd.ratingsSetup.MetaChain.ValidatorDecreaseFactor,
+			consecutiveMissedBlocksPenalty:  rd.ratingsSetup.MetaChain.ConsecutiveMissedBlocksPenalty,
+			proposerValidatorImportance:     rd.ratingsSetup.MetaChain.ProposerValidatorImportance,
+		}
+		metaRatingsStepData, err := computeRatingStep(metaRatingsStepsArgs)
+		if err != nil {
+			return fmt.Errorf("%w while computing metachain rating steps for epoch %d", err, chainParams.EnableEpoch)
+		}
+
+		ratingsStepsConfig = append(ratingsStepsConfig, ratingsStepsData{
+			enableEpoch:          chainParams.EnableEpoch,
+			shardRatingsStepData: shardRatingsStepData,
+			metaRatingsStepData:  metaRatingsStepData,
+		})
+	}
+
+	// TODO: add unit tests for the new changes in a future PR
+	sort.SliceStable(ratingsStepsConfig, func(i, j int) bool {
+		return ratingsStepsConfig[i].enableEpoch < ratingsStepsConfig[j].enableEpoch
+	})
+
+	rd.ratingsStepsConfig = ratingsStepsConfig
+
+	return nil
+}
+
+// EpochConfirmed will be called whenever a new epoch is called
+func (rd *RatingsData) EpochConfirmed(epoch uint32, _ uint64) {
+	log.Debug("RatingsData - epoch confirmed", "epoch", epoch)
+
+	rd.mutConfiguration.Lock()
+	defer rd.mutConfiguration.Unlock()
+
+	newVersion := rd.getMatchingVersion(epoch)
+	if rd.currentRatingsStepData.enableEpoch == newVersion.enableEpoch {
+		return
+	}
+
+	rd.currentRatingsStepData = newVersion
+
+	log.Debug("updated shard ratings step data",
+		"epoch", epoch,
+		"proposer increase rating step", newVersion.shardRatingsStepData.ProposerIncreaseRatingStep(),
+		"proposer decrease rating step", newVersion.shardRatingsStepData.ProposerDecreaseRatingStep(),
+		"validator increase rating step", newVersion.shardRatingsStepData.ValidatorIncreaseRatingStep(),
+		"validator decrease rating step", newVersion.shardRatingsStepData.ValidatorDecreaseRatingStep(),
+	)
+
+	log.Debug("updated metachain ratings step data",
+		"epoch", epoch,
+		"proposer increase rating step", newVersion.metaRatingsStepData.ProposerIncreaseRatingStep(),
+		"proposer decrease rating step", newVersion.metaRatingsStepData.ProposerDecreaseRatingStep(),
+		"validator increase rating step", newVersion.metaRatingsStepData.ValidatorIncreaseRatingStep(),
+		"validator decrease rating step", newVersion.metaRatingsStepData.ValidatorDecreaseRatingStep(),
+	)
+}
+
+func (rd *RatingsData) getMatchingVersion(currentEpoch uint32) ratingsStepsData {
+	var ratingsStep ratingsStepsData
+	for _, ratingsStepConfig := range rd.ratingsStepsConfig {
+		if ratingsStepConfig.enableEpoch > currentEpoch {
```

Comment:
on initializing rd.ratingsStepsConfig can you make sure there is a config for enableEpoch == 0?
otherwise you may return nil

Reply:
added TODO, will change in the next PR